homework9.js

```diff
@@ -0,0 +1,273 @@
+// # Exercises
+
+// __NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM!
+
+// 1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
+Array.prototype.myForEach = function(callback) {
+
+ for(let i = 0; i < this.length; i++) {
+  callback(this[i])
+   }
+};
+
+// [1,2,3].myForEach((el) => {
+//  console.log(el)
+// })
+
+// 2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+Array.prototype.myMap = function(callback) {
+  let answer = [];
+  this.myForEach((el) => {
+    answer.push(callback(el))
+  })
+  return answer;
+};
+
+
+
+// 3. Write a 'myFilter` that behave's the same as `filter`.
+
+Array.prototype.myFilter = function(callback) {
+  let answer = [];
+  for (let i = 0; i < this.length; i++) {
+    if (callback(this[i])) {
+      answer.push(this[i])
+    }
+  }
+  return answer;
+}
+
+// 4. Write a `myEvery`.
+
+Array.prototype.myEvery = function(callback) {
+  let result = true;
+  for (let i = 0; i < this.length; i++) {
+    if (!callback(this[i])) {
+      result = false;
+    }
+  }
+  return result;
+}
+let arr2 = [1,3,5];
+
+arr2.myEvery( el => {
+  return el % 2 !== 0
+})
+
+// 5. Write a `myReduce`. It should use the first element in the array if none is given.
+
+
+
+Array.prototype.myReduce = function (cb, acc) {
+  let startidx = 0
+  if (!acc) {
+    acc = this[0];
+    startidx++
+  }
+  for (let i = startidx; i < this.length; i++) {
+    acc = cb(acc, this[i])
+  }
+  return acc
+}
+
+let myRedArr = [1,2,3]
+
+myRedArr.myReduce((acc, el) =>
+  acc + el, 9)
+
+
+
+
+
+
+// 6. Write a `myTranspose`. This function should transpose a matrix. Exp:
+// ```js
+// let mtx = [
+//             [1, 2],
+//             [3, 4],
+//             [5, 6]
+//           ]
+//
+// mtx.myTranspose();
+//
+// // => [
+//         [1, 3, 5],
+//         [2, 4, 6]
+//        ]
+// ```
+function myTranspose (mtx) {
+
+let grid = [];
+  for (let i = 0; i < mtx[0].length; i++) {
+    grid[i] = [];
+    for (let j = 0; j < mtx.length; j++) {
+      grid[i][j] = mtx[j][i];
+    }
+  }
+  return grid;
+}
+
+
+let  mtx = [[1,2],[3,4],[5,6]];
+myTranspose(mtx)
+
+
+
+
+
+//
+// 7. Use `myMap` to build a new array where every element is incremented by 10.
+let arr4 = [1, 2, 3];
+let arrByTen = [];
```

Comment:
Ah, okay! So. I think you're mixing up myMap with myForEach a little bit here. Remember, myMap returns its own array, so you shouldn't need to push into a new one